transaction-status/src/token_balances.rs

```diff
@@ -59,25 +59,20 @@ pub fn collect_token_balances(
 
     for (_, transaction) in OrderedIterator::new(batch.transactions(), batch.iteration_order()) {
         let account_keys = &transaction.message.account_keys;
-        let mut fetch_account_hash: HashMap<u8, bool> = HashMap::new();
-        for instruction in transaction.message.instructions.iter() {
-            if let Some(program_id) = account_keys.get(instruction.program_id_index as usize) {
-                if is_token_program(&program_id) {
-                    for account in &instruction.accounts {
-                        fetch_account_hash.insert(*account, true);
-                    }
-                }
-            }
-        }
+        let has_token_program = account_keys.iter().any(|p| is_token_program(p));
 
         let mut transaction_balances: Vec<TransactionTokenBalance> = vec![];
-        for index in fetch_account_hash.keys() {
-            if let Some(account_id) = account_keys.get(*index as usize) {
+        if has_token_program {
+            for (index, account_id) in account_keys.iter().enumerate() {
+                if is_token_program(account_id) {
```

Comment:
Ah, nice, but we can go even further: we can also skip checking addresses in Message::program_ids().

Reply:
Ah, good plan!